ivadomed/loader/loader.py

```diff
@@ -1037,77 +1037,9 @@ def __init__(self, bids_df, subject_file_lst, target_suffix, contrast_params, sl
-                    # Otherwise, multiple raters segmented the same class.
-                    if isinstance(suffix_list, list):
-                        for suffix in suffix_list:
-                            if suffix in deriv:
-                                target_filename[idx].append(deriv)
-                    elif suffix_list in deriv:
-                        target_filename[idx] = deriv
-                if not (self.roi_params["suffix"] is None) and self.roi_params["suffix"] in deriv:
-                    roi_filename = [deriv]
-
-            if (not any(target_filename)) or (not (self.roi_params["suffix"] is None) and (roi_filename is None)):
-                continue
-
-            metadata = df_sub.to_dict(orient='records')[0]
-            metadata['contrast'] = contrast
-
-            if len(bounding_box_dict):
-                # Take only one bounding box for cropping
-                metadata['bounding_box'] = bounding_box_dict[str(df_sub['path'].values[0])][0]
-
-            if metadata_choice == 'mri_params':
-                if not all([imed_film.check_isMRIparam(m, metadata, subject, self.metadata) for m in
-                            self.metadata.keys()]):
-                    continue
-
-            elif metadata_choice and metadata_choice != 'contrasts' and metadata_choice is not None:
-                # add custom data to metadata
-                if metadata_choice not in df_sub.columns:
-                    raise ValueError("The following metadata cannot be found: {}. "
-                                     "Invalid metadata choice.".format(metadata_choice))
-                metadata[metadata_choice] = df_sub[metadata_choice].values[0]
-                # Create metadata dict for OHE
-                data_lst = sorted(set(bids_df.df[metadata_choice].dropna().values))
-                metadata_dict = {}
-                for idx, data in enumerate(data_lst):
-                    metadata_dict[data] = idx
-                metadata['metadata_dict'] = metadata_dict
-
-            # Fill multichannel dictionary
-            # subj_id is the filename without modality suffix and extension
-            if multichannel:
-                idx = idx_dict[df_sub['suffix'].values[0]]
-                subj_id = subject.split('.')[0].split('_')[0]
-                multichannel_subjects[subj_id]["absolute_paths"][idx] = df_sub['path'].values[0]
-                multichannel_subjects[subj_id]["deriv_path"] = target_filename
-                multichannel_subjects[subj_id]["metadata"][idx] = metadata
-                if roi_filename:
-                    multichannel_subjects[subj_id]["roi_filename"] = roi_filename
-            else:
-                self.filename_pairs.append(([df_sub['path'].values[0]],
-                                            target_filename, roi_filename, [metadata]))
+            multichannel_subjects = self.create_filename_pair(multichannel_subjects, subject, c, tot, multichannel, 
```

Comment:
I find the return of `multichannel_subjects` here hard to read/understand because `create_filename_pair` does 2 different things. 1- Returns `multichannel_subjects` when multichannel is true. 2- Fills `self.filename_pairs` when multichannel is false.